jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1162,12 +1162,22 @@ public Method getDeclaredMethod(String name, Class<?>... parameterTypes) throws
  *
  * @return		the declaring class of the receiver.
  */
+private transient Class<?> cachedDeclaringClass;
```

Comment:
Declaring the fields here disconnects the javadoc from the method it describes.  The fields should likely be declared at the top of the file with other variables.